services/motion/localizer.go

```diff
@@ -66,20 +125,29 @@ func (m *movementSensorLocalizer) CurrentPosition(ctx context.Context) (*referen
 	if err != nil {
 		return nil, err
 	}
+
+	if err := validateGeopoint(gp); err != nil {
+		return nil, err
+	}
 	var o spatialmath.Orientation
 	properties, err := m.Properties(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
+
 	switch {
 	case properties.CompassHeadingSupported:
 		headingLeft, err := m.CompassHeading(ctx, nil)
 		if err != nil {
 			return nil, err
 		}
+
+		if math.IsNaN(headingLeft) {
+			return nil, errors.New("heading can't be NaN")
+		}
 		// CompassHeading is a left-handed value. Convert to be right-handed. Use math.Mod to ensure that 0 reports 0 rather than 360.
-		heading := math.Mod(math.Abs(headingLeft-360), 360)
-		o = &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: heading}
+		theta := rdkutils.SwapCompassHeadingHandedness(headingLeft)
```

Comment:
I don't think this is super necessary, the comment and the math make it pretty clear without it being a different function